discojs-web/src/memory/memory.ts

```diff
@@ -10,62 +10,72 @@ import { Map } from 'immutable'
 import * as tf from '@tensorflow/tfjs'
 
 import type { Path, Model, ModelInfo, ModelSource } from '@epfml/discojs'
-import { Memory, ModelType, models } from '@epfml/discojs'
+import { Memory, StoredModelType, models } from '@epfml/discojs'
 
 export class IndexedDB extends Memory {
-  override pathFor (source: ModelSource): Path {
+  override getModelMemoryPath (source: ModelSource): Path {
     if (typeof source === 'string') {
       return source
     }
-
-    if (source.type === undefined || source.taskID === undefined || source.name === undefined) {
+    if (source.type === undefined || source.taskID === undefined ||
+      source.name === undefined || source.tensorBackend === undefined) {
```

Comment:
`ModelSource` has only `type` & `version` as potentially undef, no need to check for everyone

```suggestion
    if (source.type === undefined) {
```

Reply:
I'll actually make `ModelInfo.type` required, I don't see any uses of it being left undefined right now